### PR TITLE
Fix spec-compliance of Administrator Commissioning attributes.

### LINKED ISSUE #20882: AdministratorCommissioning cluster not implemented per spec
#### Problem
Several problems:
1. WindowStatus does not return WindowNotOpen during initial commissioning, as far as I can tell.  It should.
2. AdminFabricIndex and AdminVendorId don't return null when they should (i.e. any time status is WindowNotOpen).

Depending on how https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/5473 is resolved that could be an issue too.

 #### Proposed Solution
Add tests, align with spec.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
* Move management of the AdminVendorId and AdminFabricIndex to the
  commissioning window opener.
* Fix the types of the attributes in the XML to be nullable.
* Implement the spec provisions about returning null for the fabric
  index when the fabric gets removed.
* Implement the spec provisions about returning "window not open" when
  it was opened via non-cluster means (e.g. boot time.

Fixes https://github.com/project-chip/connectedhomeip/issues/20882

#### Problem
See above.

#### Change overview
See above.

#### Testing
Added YAML and unit tests, plus manually tested that during initial PASE the attributes return the right values now.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/app/server/CommissioningWindowManager.cpp [hunk 1]
```diff
@@ -324,6 +347,31 @@ void CommissioningWindowManager::CloseCommissioningWindow()
...
+bool CommissioningWindowManager::IsCommissioningWindowOpen() const
+{
+    return mWindowStatus != CommissioningWindowStatus::kWindowNotOpen;
+}
+
+void CommissioningWindowManager::OnFabricRemoved(FabricIndex removedIndex)
+{
+    if (!mOpenerFabricIndex.IsNull() && mOpenerFabricIndex.Value() == removedIndex)
+    {
+        // Per spec, we shoiuld clear out the stale fabric index.
```
Comment (tcarmelveilleux) [inline src/app/server/CommissioningWindowManager.cpp:370]: nit: `should clear`

File: src/credentials/FabricTable.h [hunk 2]
```diff
@@ -104,7 +104,7 @@ class DLL_EXPORT FabricInfo
         return CHIP_NO_ERROR;
     }
 
-    uint16_t GetVendorId() const { return mVendorId; }
+    VendorId GetVendorId() const { return mVendorId; }
```
Comment (tcarmelveilleux) [inline src/credentials/FabricTable.h:107]: The reason it was uint16_t was that VendorId, in SDK, is an *enum* and it's very unclean to have an enum use as a full-width type but only have a few values defined. We should make VendorId uint16_t and make WellKnownVendorId enum be of type VendorId.

**Review by tcarmelveilleux — APPROVED** [review]

File: src/app/server/CommissioningWindowManager.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/server/CommissioningWindowManager.cpp:370]: Good catch, fixed.

File: src/credentials/FabricTable.h (hunk 2 above)
Comment (bzbarsky-apple) [reply src/credentials/FabricTable.h:107]: Filed https://github.com/project-chip/connectedhomeip/issues/21459

**Pull request merged**